File: dotnet/Samples/LinguisticOptions/LinguisticOptionsService.cs
﻿using LanguageWeaver.Sdk.Common;
using LanguageWeaver.Sdk.Configurations;
using LanguageWeaver.Sdk.LinguisticOptions.Common.Request;
using LanguageWeaver.Sdk.LinguisticOptions.Common.Result;

namespace LanguageWeaver.Sdk.Samples.LinguisticOptions;

public class LinguisticOptionsService
{
    public static void Main()
    {
        using var lwClient = new SdkFactory().GetLanguageWeaverClient(new ClientConfiguration());

        GetLinguisticOptionsRequest getLinguisticOptionsRequest = new GetLinguisticOptionsRequest()
        {
            SourceLanguageId = "chi",
            TargetLanguageId = "eng",
            Model = "generic"
        };

        LinguisticOptionsResult linguisticOptionsResult = lwClient.GetLinguisticOptions(getLinguisticOptionsRequest);
        //handle result
    }
}

File: javascript/backend/samples/linguisticOptions/LinguisticOptionsService.js
const SdkFactory = require("@language-weaver/lw-sdk-js").default;
const {ClientConfiguration, GetLinguisticOptionsRequest} = require("@language-weaver/lw-sdk-js");

const getLinguisticOptions = async () => {
    try {
        const lwClient = await SdkFactory.getLanguageWeaverClient(new ClientConfiguration());
        const request = new GetLinguisticOptionsRequest();
        request.sourceLanguageId = "chi";
        request.targetLanguageId = "eng";
        request.model = "generic";
        const linguisticOptionsResult = await lwClient.getLinguisticOptions(request);
        console.log(linguisticOptionsResult.linguisticOptions);
    } catch (e) {
        console.log(e);
    }
}

getLinguisticOptions();